sos/collector/transports/saltstack.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Red Hat 2022, Trevor Benson <trevor.benson@gmail.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import contextlib
+import json
+import os
+import shutil
+from sos.collector.transports import RemoteTransport
+from sos.collector.exceptions import ConnectionException
+from sos.utilities import (is_executable,
+                           sos_get_command_output)
+
+
+class SaltStackMasterTransport(RemoteTransport):
+    """
+    A transport for collect that leverages SaltStack's Master Pub/Sub
+    functionality to send commands to minions.
+
+    This transport will by default assume the use cmd.shell module to
+    execute commands on the minions.
+    """
+
+    name = 'saltstack'
+
+    def _convert_output_json(self, json_output):
+        return list(json.loads(json_output).values())[0]
+
+    def run_command(
+            self, cmd, timeout=180, need_root=False, env=None, get_pty=False):
+        """
+        Run a command on the remote host using SaltStack Master.
+        If the output is json, convert it to a string.
+        """
+        ret = super(SaltStackMasterTransport, self).run_command(
+            cmd, timeout, need_root, env, get_pty)
+        with contextlib.suppress(Exception):
+            ret['output'] = self._convert_output_json(ret['output'])
+        return ret
+
+    def _salt_retrieve_file(
+            self, node, fname, dest,
+            timeout=180, need_root=False, env=None, get_pty=False):
```

Comment:
Sorry, missed this earlier. With current use of `sos_get_command_output()`, the parameters after `timeout` are unnecessary and can be removed. `timeout` can be as well, unless we want to use that for the `cp.push` command instead of the default 300 seconds.

After this, everything looks good to me for merge.

Reply:
Relying on default of 300 seconds for rare cases where network is slow and log collection has balooned sos report size.